pygmt/base_plotting.py

```diff
@@ -1922,3 +1922,240 @@ def update_pointers(data_pointers):
             with file_context as fname:
                 arg_str = " ".join([fname, build_arg_string(kwargs)])
                 lib.call_module("meca", arg_str)
+
+    @fmt_docstring
+    @use_alias(
+        A="vector",
+        B="frame",
+        C="cmap",
+        D="rescale",
+        E="uncertainty_color",
+        G="facecolor",
+        J="projection",
+        L="line",
+        N="no_clip",
+        R="region",
+        S="scaling",
+        U="timestamp",
+        V="verbose",
+        W="pen",
+        X="xshift",
+        Y="yshift",
+        p="perspective",
+        t="transparency",
+    )
+    @kwargs_to_strings(R="sequence", i="sequence_comma")
+    def velo(self, data=None, scaling=None, **kwargs):
```

Comment:
I thought leaving `scaling` in the function definition would highlight the fact that it is a required argument and not an optional argument (https://docs.generic-mapping-tools.org/dev/supplements/geodesy/velo.html). 

But maybe it is another issue., and it should be consistent for the other required arguments (e.g. `projection` and `region`)

Reply:
What about removing it from the alias? as I did in my last commit https://github.com/GenericMappingTools/pygmt/pull/525/commits/7a9de97f3236dd85e280287484430eb80545979c

Reply:
I just realised it is not so simple. Any suggestion @weiji14 about how to move foward if I want to remove the alias but keep the argument `scaling`? What command should I add? I thought the `extra_arrays.append(scaling)` would do the job:
```
        if scaling is not None:
            if not isinstance(scaling,str):
                raise GMTInvalidInput(
                    "Scaling is a required argument and has to be a string."
                )
            extra_arrays.append(scaling)
```

Reply:
I would keep the alias for S="scaling" (i.e. revert https://github.com/GenericMappingTools/pygmt/commit/7a9de97f3236dd85e280287484430eb80545979c). Don't think we need to explicitly check that the user passed in `scaling` as an argument, as it will become very apparent when GMT errors out.

Reply:
Thanks @weiji14 , I will do that. I guess you are right.  Maybe it is just me, I quite like seeing the required arguments written as arguments in the signature of the function and optional arguments as kwargs. I think it helps new users who are not familiar with the function, particularly when they display the help, particularly in jupyter notebook. I may open an issue so we can discuss if it something that could bring more clarity to new users who are not familiar with the requirred argument in gmt

<img width="1341" alt="Screenshot 2020-11-13 at 12 15 19" src="https://user-images.githubusercontent.com/10154151/99071488-2035d900-25aa-11eb-9ae0-1c36b8811be6.png">

Reply:
Actually you can keep both. Have the alias S="scaling" and also list it in the `def velo(scaling=None):` part. Just that the latter will require some extra code to check for stuff :slightly_smiling_face:

Reply:
@weiji14 But if we go this way, I am wondering if it would not make more sense to also add `region=none` and `projection=none` as they are also required argument... What are your thoughts?

Reply:
`region` and `projection` are only required if it's the first `fig`, they could be set e.g. by `fig.basemap()`. So just having `scaling=None` should be sufficient.